contrib/bloop/src/mill.contrib.bloop/BloopModule.scala

```diff
@@ -0,0 +1,197 @@
+    mkdir(bloopDir)
+
+    val modules = if (ev != null) {
+      val rootModule = ev.rootModule
+      rootModule.millInternal.segmentsToModules.values.collect {
+        case m: scalalib.JavaModule => m
+      }.toSeq
+    } else Seq()
+
+    Task.traverse(modules)(writeBloopConfig(bloopDir, _))
+  }
+
+  implicit def millScoptEvaluatorReads[T] = new mill.main.EvaluatorScopt[T]()
+
+  def bloopConfig(bloopDir: Path,
+                  module: JavaModule): Task[(BloopFile, Path)] = {
+    import _root_.bloop.config.Config
+    def name(m: JavaModule) = m.millModuleSegments.render
+    def out(m: JavaModule) = bloopDir / "out" / m.millModuleSegments.render
+    def classes(m: JavaModule) = out(m) / "classes"
+
+    val javaConfig =
+      module.javacOptions.map(opts => Some(Config.Java(options = opts.toList)))
+    val scalaConfig = module match {
+      case s: ScalaModule =>
+        T.task {
+          val pluginOptions = s.scalacPluginClasspath().map { pathRef =>
+            s"-Xplugin:${pathRef.path}"
+          }
+
+          Some(
+            Config.Scala(
+              organization = "org.scala-lang",
+              name = "scala-compiler",
+              version = s.scalaVersion(),
+              options = (s.scalacOptions() ++ pluginOptions).toList,
+              jars = s.scalaCompilerClasspath().map(_.path.toNIO).toList,
+              analysis = None,
+              setup = None
+            )
+          )
+        }
+      case _ => T.task(None)
+    }
+
+    val platform = T.task {
+      Config.Platform.Jvm(
+        Config.JvmConfig(
+          home = T.ctx().env.get("JAVA_HOME").map(s => Path(s).toNIO),
+          options = module.forkArgs().toList
+        ),
+        mainClass = module.mainClass()
+      )
+    }
+
+    val testConfig = module match {
+      case m: TestModule =>
+        T.task {
+          Some(
+            Config.Test(
+              frameworks = m
+                .testFrameworks()
+                .map(f => Config.TestFramework(List(f)))
+                .toList,
+              options = Config.TestOptions(
+                excludes = List(),
+                arguments = List()
+              )
+            )
+          )
+        }
+      case _ => T.task(None)
+    }
+
+    val ivyDepsClasspath =
+      module
+        .resolveDeps(T.task {
+          module.compileIvyDeps() ++ module.transitiveIvyDeps()
+        })
+        .map(_.map(_.path).toSeq)
+
+    def transitiveClasspath(m: JavaModule): Task[Seq[Path]] = T.task {
+      m.moduleDeps.map(classes) ++
+        m.resources().map(_.path) ++
+        m.unmanagedClasspath().map(_.path) ++
+        Task.traverse(m.moduleDeps)(transitiveClasspath)().flatten
+    }
+
+    val classpath = T.task(transitiveClasspath(module)() ++ ivyDepsClasspath())
+
+    val project = T.task {
+      Config.Project(
+        name = name(module),
+        directory = module.millSourcePath.toNIO,
+        sources = module.sources().map(_.path.toNIO).toList,
```

Comment:
Here's the offender (I think). See my [comment](https://github.com/Baccata/mill/pull/1/files#r218299266)